woocommerce/woocommerce
# Adding attribute edit modal for products MVP

### All Submissions:






### Changes proposed in this Pull Request:

Adding the edit modal for attributes as per [the design](Show or hide this attribute in the filters section on your store’s category and shop pages).

![image](https://user-images.githubusercontent.com/444632/198140693-d8309fb0-b3c6-495b-9438-cf43341228fa.png)

**Note:** One complicating issue with managing attributes has to do with the product API endpoint only including the _names_ of the various terms, which are not unique. To address this we're now fetching the terms associated with the product from the product terms endpoint, and using them to manage the adding/editing and various functionality, updating the global form data as needed. These are stored as the `hydratedAttributes` with the `attribute-field` component. I've now refactored the add-modal to address these changes as well.

Closes #34332 .






### How to test the changes in this Pull Request:



1. Checkout branch
2. Ensure you have at least 2 attributes with options configured.
3. Enable new products MVP feature flag
4. Go to new products MVP screen
5. Click the "add attribute button" at the bottom to open the add modal.
6. Add one of the attributes, as well as 1 or two of the associated options.
7. Click "Add" button, and ensure the attribute shows up in the attribute table.
8. Click "Edit" button on added attribute to open the edit modal.
9. Add another option to the attribute and click "update."
10. Ensure the UI on the attribute section has updated with that attribute.
11. Publish the product, but don't navigate away
12. Refresh the page, which should just load the same product.
13. Scroll to the bottom and ensure the updated options were persisted.
14. Add another attribute with the add modal, but don't publish the product.
15. Click "edit" on one of the attributes to open the edit modal.
16. Delete the name of the attribute in the top "attribute" field, and select the other from the drop-down.
17. It should switch the attribute field and auto populate the terms in the lower drop-down associated with this attribute.
18. Similarly, test delete attributes and checking both checkboxes, publish the product, and ensure the changes were persisted.



### Other information:




### FOR PR REVIEWER ONLY:
